project-addons/purchase_picking/views/purchase_view.xml

```diff
@@ -19,6 +19,30 @@
                         attrs="{'invisible': [('state','in',('draft', 'cancel'))]}"
                         class="oe_highlight"/>
             </button>
+            <xpath expr="//field[@name='order_line']/tree//field[@name='name']" position="attributes">
+                <attribute name="attrs">{'readonly': [('state', '=', 'purchase')]}</attribute>
```

Comment:
Si queréis qeu una vez confirmada la compra no sea editable, hay un parámetro para hacrlo, creo que estos cambios no serían necesarios, el paámetro está en /Compras/Comparas/Configuración y se llama "Bloquear pedidos confirmados"

Reply:
Desconocía la existencia del parámetro, lo he probado pero no es exactamente lo que queremos, queremos que en el estado "Pedido de compra" las líneas no sean editables y el parámetro pasa el pedido a estado "Bloqueado" cuando se confirma. En cualquier caso, tenemos pendiente revisar los estados tanto de pedido de compra y de venta para ver cómo lo quieren realmente, tendremos en mente este parámetro que parece interesante.

Reply:
entonces lo dejáis así por ahora? apruebo el PR?

Reply:
Creo que de momento si, por que teníamos algunos problemillas de que editaban cosas en ese estado y no deberían.
Como te digo lo volveremos a mirar y seguramente este código desaparezca. Pero de momento nos quedamos con él.